components/director/internal/domain/token/service_test.go

```diff
@@ -0,0 +1,164 @@
+
+		//WHEN
+		_, err := svc.GenerateOneTimeToken(ctx, runtimeID, model.RuntimeReference)
+
+		//THEN
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), testErr.Error())
+		cli.AssertExpectations(t)
+		sysAuthSvc.AssertExpectations(t)
+	})
+
+	t.Run("Error - saving auth failed", func(t *testing.T) {
+		ctx := context.TODO()
+		testErr := errors.New("test error")
+		cli := &automock.GraphQLClient{}
+		sysAuthSvc := &automock.SystemAuthService{}
+		sysAuthSvc.On("Create", ctx, model.RuntimeReference, runtimeID, (*model.AuthInput)(nil)).
+			Return("", testErr)
+		svc := token.NewTokenService(cli, sysAuthSvc, URL)
+
+		//WHEN
+		_, err := svc.GenerateOneTimeToken(ctx, runtimeID, model.RuntimeReference)
+
+		//THEN
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), testErr.Error())
+		cli.AssertExpectations(t)
+		sysAuthSvc.AssertExpectations(t)
+	})
+}
+
+func TestTokenService_GetOneTimeTokenForApp(t *testing.T) {
+	appID := "98cb3b05-0f27-43ea-9249-605ac74a6cf0"
+	authID := "77cabc16-9fb8-4338-b252-7b404f2e6487"
+
+	expectedRequest := gcli.NewRequest(fmt.Sprintf(`
+		mutation { generateApplicationToken (appID:"%s")
+		  {
+			token
+		  }
+		}`, authID))
+
+	t.Run("Success application token", func(t *testing.T) {
+		//GIVEN
+		ctx := context.TODO()
+		cli := &automock.GraphQLClient{}
+		expectedToken := "token"
+
+		expected := token.ExternalTokenModel{AppToken: token.ExternalRuntimeToken{Token: expectedToken}}
+		input := token.ExternalTokenModel{}
+		cli.On("Run", ctx, expectedRequest, &token.ExternalTokenModel{}).
+			Run(mockReturnFilledToken(t, input, expected)).Return(nil).Once()
+		sysAuthSvc := &automock.SystemAuthService{}
+		sysAuthSvc.On("Create", ctx, model.ApplicationReference, appID, (*model.AuthInput)(nil)).
+			Return(authID, nil)
+		svc := token.NewTokenService(cli, sysAuthSvc, URL)
+
+		//WHEN
+		authToken, err := svc.GenerateOneTimeToken(ctx, appID, model.ApplicationReference)
+
+		//THEN
+		require.NoError(t, err)
+		assert.Equal(t, expectedToken, authToken.Token)
+		cli.AssertExpectations(t)
+		sysAuthSvc.AssertExpectations(t)
+	})
+
+	t.Run("Error - generating token failed", func(t *testing.T) {
+		ctx := context.TODO()
+		cli := &automock.GraphQLClient{}
+		testErr := errors.New("test error")
+		cli.On("Run", ctx, expectedRequest, &token.ExternalTokenModel{}).
+			Return(testErr).Once()
+		sysAuthSvc := &automock.SystemAuthService{}
+		sysAuthSvc.On("Create", ctx, model.ApplicationReference, appID, (*model.AuthInput)(nil)).
+			Return(authID, nil)
+		svc := token.NewTokenService(cli, sysAuthSvc, URL)
+
+		//WHEN
+		_, err := svc.GenerateOneTimeToken(ctx, appID, model.ApplicationReference)
+
+		//THEN
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), testErr.Error())
+		cli.AssertExpectations(t)
+		sysAuthSvc.AssertExpectations(t)
+	})
+}
+
+func mockReturnFilledToken(t *testing.T, input, output token.ExternalTokenModel) func(args mock.Arguments) {
```

Comment:
my proposal for renaming:
```
func generateFakeToken(t *testing.T, generated token.ExternalTokenModel) func(args mock.Arguments) {
	return func(args mock.Arguments) {
		arg, ok := args.Get(2).(*token.ExternalTokenModel)
		require.True(t, ok)
		*arg = generated
	}
}
```
I removed also some assertions to make it clear that this function is not responsible for making assertions but ratrher faking token generation